sdk-java/src/main/java/ly/count/sdk/java/internal/ModuleViews.java

```diff
@@ -1,9 +1,497 @@
+        }
+
+        vd.viewStartTimeSeconds = TimeUtils.uniqueTimestampS();
+    }
+
+    void stopAllViewsInternal(Map<String, Object> viewSegmentation) {
+        L.d("[ModuleViews] stopAllViewsInternal");
+
+        autoCloseRequiredViews(true, viewSegmentation);
+    }
+
+    private ViewData validateViewID(String viewID, String function) {
+        if (viewID == null || viewID.isEmpty()) {
+            L.e("[ModuleViews] validateViewID, " + function + ", Trying to process view with null or empty view ID, ignoring request");
+            return null;
+        }
+
+        if (!viewDataMap.containsKey(viewID)) {
+            L.w("[ModuleViews] validateViewID, " + function + ",  there is no view with the provided view id");
+            return null;
+        }
+
+        ViewData vd = viewDataMap.get(viewID);
+        if (vd == null) {
+            L.e("[ModuleViews] validateViewID, " + function + ",  view id:[" + viewID + "] has a 'null' value. This should not be happening");
+        }
+
+        return vd;
+    }
+
+    private String validateViewWithName(String viewName, String function) {
+        if (viewName == null || viewName.isEmpty()) {
+            L.e("[ModuleViews] " + function + ", Trying to process the view with null or empty view name, ignoring request");
+            return null;
+        }
+
+        String viewID = null;
+
+        for (Map.Entry<String, ViewData> entry : viewDataMap.entrySet()) {
+            ViewData vd = entry.getValue();
+            if (vd != null && viewName.equals(vd.viewName)) {
+                viewID = entry.getKey();
+            }
+        }
+
+        if (viewID == null) {
+            L.e("[ModuleViews] " + function + ", No view entry found with the provided name :[" + viewName + "]");
+        }
+
+        return viewID;
+    }
+
+    void addSegmentationToViewWithNameInternal(@Nullable String viewName, @Nullable Map<String, Object> viewSegmentation) {
+        String viewID = validateViewWithName(viewName, "addSegmentationToViewWithNameInternal");
+        if (viewID == null) {
+            return;
+        }
+        addSegmentationToViewWithIDInternal(viewID, viewSegmentation);
+    }
+
+    private void addSegmentationToViewWithIDInternal(String viewID, Map<String, Object> viewSegmentation) {
+        ViewData vd = validateViewID(viewID, "addSegmentationToViewWithIdInternal");
+        if (vd == null) {
+            return;
+        }
+
+        if (viewSegmentation == null || viewSegmentation.isEmpty()) {
+            L.e("[ModuleViews] addSegmentationToViewWithIdInternal, Trying to record view with null or empty view segmentation, ignoring request");
+            return;
+        }
+        removeReservedKeysFromViewSegmentation(viewSegmentation);
+        vd.viewSegmentation.putAll(viewSegmentation);
+    }
+
+    /**
+     * Creates a crypto-safe SHA-256 hashed random value
+     *
+     * @return returns a random string value
+     */
+    public static String safeRandomVal() {
```

Comment:
do you use this somewhere else?

Reply:
Nope, but maybe it can be moved to a utils class. Maybe it will be needed in the future